packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -119,12 +120,13 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler, Plug
   override fun onDetachedFromActivity() {}
 
   override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
-      if (requestCode == AWSCognitoAuthPlugin.WEB_UI_SIGN_IN_ACTIVITY_CODE) {
-        Amplify.Auth.handleWebUISignInResponse(data)
-        return true
-      }  else {
-        return false
-      }
+    var isHostedUIActivity = isRedirectActivityDeclared(context)
+    if (!isHostedUIActivity && requestCode == AWSCognitoAuthPlugin.WEB_UI_SIGN_IN_ACTIVITY_CODE) {
+      Amplify.Auth.handleWebUISignInResponse(data)
+      return true
+    }  else {
+      return false
+    }
```

Comment:
Nit: Suggested minor improvement on existing code
```suggestion
    }
    return false
```

Reply:
Fixed on my end.